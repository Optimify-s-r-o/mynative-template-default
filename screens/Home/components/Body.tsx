import {Button, Text, View} from "react-native";
import {styles} from "../styles";
import {useNavigation} from "@react-navigation/native";

export const Body = () => {
    const navigator = useNavigation();

    const openModal = () => {
        navigator.navigate("Modal")
    }

    const signIn = () => {
        navigator.navigate("Portal")
    }

    return (<View style={styles.header}>
        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeaderView>
         **/}

        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeBody>
         **/}
        <Text>MyNative Body</Text>
        <Button onPress={openModal} title={"Open Modal"}/>
        <Button onPress={signIn} title={"Portal"}/>
        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         </@MyNativeBody>
         **/}
        <Text>MyNative Footer</Text>
        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeaderView/>
         **/}
    </View>);
}