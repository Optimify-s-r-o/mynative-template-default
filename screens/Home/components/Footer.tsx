import { Text, View } from 'react-native';
import React from 'react';
import { styles } from '../styles';

export const Footer = () => {
  return (
    <View style={styles.header}>
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeaderView>
         **/}

      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeFooter>
         **/}
      <Text>Footer</Text>
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         </@MyNativeFooter>
         **/}

      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeaderView/>
         **/}
    </View>
  );
};
