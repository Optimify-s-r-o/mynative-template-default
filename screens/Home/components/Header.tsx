import {Text, View} from "react-native";
import {styles} from "../styles";

export const Header = () => {
    return (<View style={styles.header}>
        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeaderView>
         **/}

        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeader>
         **/}
        <Text>Header</Text>
        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         </@MyNativeHeader>
         **/}

        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeHeaderView/>
         **/}
    </View>);
}