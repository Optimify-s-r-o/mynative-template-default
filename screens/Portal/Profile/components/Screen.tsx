import { Text, View } from 'react-native';
import React from 'react';
import { styles } from '../styles';

export const Screen = () => {
  return (
    <View style={styles.container}>
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView>
         **/}
      <Text>Profile</Text>
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView/>
         **/}
    </View>
  );
};
