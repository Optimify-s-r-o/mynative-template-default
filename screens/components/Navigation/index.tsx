import { useNavigation } from '@react-navigation/native';
import { Text, TouchableOpacity } from 'react-native';
import React from 'react';

export const Navigation = ({ title, to }: { title: string; to: any }) => {
  const navigator = useNavigation();

  const navigateTo = () => {
    navigator.navigate(to);
  };
  return (
    <TouchableOpacity onPress={navigateTo}>
      <Text>{title}</Text>
    </TouchableOpacity>
  );
};
