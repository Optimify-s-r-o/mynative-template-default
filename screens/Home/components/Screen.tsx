import {View} from "react-native";
import {styles} from "../styles";
import {Header} from "./Header";
import {Body} from "./Body";
import {Footer} from "./Footer";

export const Screen = () => {
    return (<View style={styles.container}>
        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView>
         **/}

        <Header/>
        <Body/>
        <Footer/>

        {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView/>
         **/}
    </View>);
}