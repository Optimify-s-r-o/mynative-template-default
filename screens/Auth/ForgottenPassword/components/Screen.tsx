import { Text, View } from 'react-native';
import React from 'react';
import { styles } from '../styles';
import { Navigation } from 'screens/components/Navigation';
import { Routes } from 'navigation/routes';

export const Screen = () => {
  return (
    <View style={styles.container}>
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView>
         **/}
      <Text>ForgottenPassword</Text>

      <Navigation to={Routes.SignUp} title={'SignUp'} />

      <Navigation to={Routes.SignIn} title={'SignIn'} />
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView/>
         **/}
    </View>
  );
};
