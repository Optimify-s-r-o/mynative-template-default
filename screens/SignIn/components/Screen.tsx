import React from 'react';
import { Text, View } from 'react-native';
import { styles } from '../styles';
import { Routes } from 'navigation/routes';
import { Navigation } from 'screens/components/Navigation';
import { useThemeContext } from 'context/Theme/ThemeContext';
import { TextInput } from 'plugins/MyNativeForm/components/TextInput';
import { Form } from 'plugins/MyNativeForm/components/Form';

export const Screen = () => {
  const theme = useThemeContext();
  const onSubmit = (data: any) => {
    alert(JSON.stringify(data));
    // TODO your implementation
  };

  return (
    <View style={[theme.styleSheet.box, styles.container]}>
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView>
         **/}

      <Text>SignIn</Text>

      <Form
        fields={{
          email: {
            default: '',
            render: (props) => (
              <TextInput
                label="Email"
                type="email"
                {...props}
                rules={{
                  required: 'You must enter valid email address.',
                  pattern: {
                    value:
                      /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
                    message: 'You must enter valid email address.',
                  },
                }}
              />
            ),
          },
          password: {
            default: '',
            render: (props) => (
              <TextInput
                label="Password"
                type="password"
                {...props}
                rules={{ required: 'You must enter password.' }}
              />
            ),
          },
        }}
        onSubmit={onSubmit}
      />

      <Navigation to={Routes.SignUp} title={'SignUp'} />

      <Navigation to={Routes.ForgottenPassword} title={'ForgottenPassword'} />

      <Navigation to={Routes.Portal} title={'Portal'} />
      {/**
         Please don't delete this comment block.
         Deleting this comment block will cause non-functionality of MyNative.
         <@MyNativeView/>
         **/}
    </View>
  );
};
