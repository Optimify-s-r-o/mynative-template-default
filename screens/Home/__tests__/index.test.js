import * as React from 'react';
import renderer from 'react-test-renderer';
import { Footer } from '../components/Footer';

it(`homePage renders correctly`, () => {
  const tree = renderer.create(<Footer />).toJSON();

  expect(tree).toMatchSnapshot();
});
