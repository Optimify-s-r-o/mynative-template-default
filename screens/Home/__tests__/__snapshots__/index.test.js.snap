// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePage renders correctly 1`] = `
<View
  style={Object {}}
>
  <Text>
    Footer
  </Text>
</View>
`;
